clippy_lints/src/strings.rs

```diff
@@ -173,16 +176,63 @@ fn is_add(cx: &LateContext<'_>, src: &Expr<'_>, target: &Expr<'_>) -> bool {
     }
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Check if the string is transformed to byte array and casted back to string.
+    ///
+    /// **Why is this bad?** It's unnecessary, the string can be used directly.
+    ///
+    /// **Known problems:** None
+    ///
+    /// **Example:**
+    /// ```rust
+    /// let string = "Hello World!";
+    /// let string2 = std::str::from_utf8(&string.as_bytes()[6..11]).unwrap();
+    /// ```
+    /// could be written as
+    /// ```rust
+    /// let string = "Hello World!";
+    /// let string2 = &string[6..11];
+    /// ```
+    pub STRING_FROM_UTF8_AS_BYTES,
+    correctness,
+    "make sure that access from slice should be done from bytes and not from string directly"
```

Comment:
This seems confusing. How about `"casting string slices to byte slices and back"`?

Reply:
Done